echochar.c

```diff
@@ -40,17 +44,108 @@ void echochar(uint8_t c) {
 			case 1:
 				switch (c) {
 					case 127: printf(" "); break;
+					/* ISO8859-15 */
+/* Generated by X16 BASIC V2:
+10 FOR I=$A0 TO $FF
+20 PRINT "case";I;CHR$($9D)": printf(";CHR$(34);CHR$(I);CHR$(34);"); break;"
+30 NEXT
+*/
+					case 160: printf(" "); break;
```

Comment:
For 88 of these 96 values, the code on the left is the code on the right anyway. You have just made your text editor pre-convert the output into UTF-8. As I have stated before, for the codes $A0-$FF (except for 8 values), Unicode is the same as 8859-15, so can just pass the values into a (quite simple) UTF-8 encoder, instead of having a huge table of pre-encoded UTF-8 values.